docs/website/versioned_docs/version-3.0.0/command-reference/dev.md

```diff
@@ -0,0 +1,226 @@
+
+Afterwards, run `odo dev`:
+
+```sh
+$ odo dev
+  __
+ /  \__     Developing using the my-nodejs-app Devfile
+ \__/  \    Namespace: default
+ /  \__/    odo version: v3.0.0-alpha1
+ \__/
+
+↪ Deploying to the cluster in developer mode
+ ✓  Waiting for Kubernetes resources [3s]
+ ✓  Syncing files into the container [335ms]
+ ✓  Building your application in container on cluster [2s]
+ ✓  Executing the application [1s]
+
+Your application is now running on the cluster
+ - Forwarding from 127.0.0.1:40001 -> 3000
+
+Watching for changes in the current directory /Users/user/nodejs
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+```
+
+In the above example, three things have happened:
+  * Your application has been built and deployed to the cluster
+  * `odo` has port-forwarded your application for local accessability
+  * `odo` will watch for changes in the current directory and rebuild the application when changes are detected
+
+## Devfile (Advanced Usage)
+
+
+### Description
+
+When `odo dev` is ran, it first looks in the `devfile.yaml` for the instructions to be executed, specifically: `components` and `commands`.
+
+Each command has a group `kind` key which correspond to either: `build`, `run`, `test`, or `debug`.
+
+These instructions make up the development cycle of `odo dev`.
+
+### Snippets
+
+With the following example `devfile.yaml` file generated by `odo init` and selecting `nodejs`, a container image will be pushed to the cluster, as well as your source code in order to start the development inner loop cycle.
+
+A much more descriptive explanation on each part of a Devefile can be found on the [Devfile API reference](https://devfile.io/docs/devfile/2.0.0/user-guide/api-reference/) site.
+
+Below, we'll explain each section of the corresponding Devfile:
+
+#### `metadata`
+
+> Descriptive registry information with regards to the devfile being used.
+```yaml
+metadata:
+  description: Stack with Node.js 14
+  displayName: Node.js Runtime
+  icon: https://nodejs.org/static/images/logos/nodejs-new-pantone-black.svg
+  language: javascript
+  name: my-nodejs-app
+  projectType: nodejs
+  tags:
+  - NodeJS
+  - Express
+  - ubi8
+  version: 1.0.1
+schemaVersion: 2.0.0
+```
+
+#### `starterProjects`
+> The starter project that was initialized when running `odo init`.
+```yaml
+starterProjects:
+- git:
+    remotes:
+      origin: https://github.com/odo-devfiles/nodejs-ex.git
+  name: nodejs-starter
+```
+
+#### `commands`
+
+> The list of commands to be used when running `odo dev`.
+>
+> Each command contains what will be run in the command line as well as what corresponding group they are part of.
+>
+> The four groups are: build, run, test and debug.
+>
+> Build is what is initially ran when deploying to the cluster.
+>
+> Run executes the command when the container has been created.
+>
+> Debug is enabled when running `odo dev --debug`.
```

Comment:
Seems straightforward, but it could be interesting to indicate that these commands are the commands needed to - build the program, - execute it in non-debug mode, - in debug mode, and - execute the tests.